libraries/griptape_nodes_library/griptape_nodes_library/execution/for_each_start.py

```diff
@@ -0,0 +1,136 @@
+from typing import Any
+
+from griptape_nodes.exe_types.core_types import (
+    ControlParameterInput,
+    ControlParameterOutput,
+    Parameter,
+    ParameterList,
+    ParameterMode,
+    ParameterTypeBuiltin,
+)
+from griptape_nodes.exe_types.flow import ControlFlow
+from griptape_nodes.exe_types.node_types import BaseNode, EndLoopNode, StartLoopNode
+
+
+class ForEachStartNode(StartLoopNode):
+    """For Each Start Node that runs a connected flow for each item in a parameter list.
+
+    This node iterates through each item in the input list and runs the connected flow for each item.
+    It provides the current item to the next node in the flow and keeps track of the iteration state.
+    """
+
+    _items: list[Any]
+    _flow: ControlFlow | None = None
+
+    def __init__(self, name: str, metadata: dict[Any, Any] | None = None) -> None:
+        super().__init__(name, metadata)
+        self.finished = False
+        self.current_index = 0
+        self._items = []
+        self.exec_out = ControlParameterOutput(tooltip="Continue the flow", name="exec_out")
+        self.add_parameter(self.exec_out)
+        self.exec_in = ControlParameterInput()
+        self.add_parameter(self.exec_in)
+        self.index_count = Parameter(
+            name="index",
+            tooltip="Current index of the iteration",
+            type=ParameterTypeBuiltin.INT.value,
+            allowed_modes={ParameterMode.PROPERTY, ParameterMode.OUTPUT},
+            settable=False,
+            default_value=0,
+        )
+        self.add_parameter(self.index_count)
```

Comment:
Can we move the index count so it appears after the item? that'll be more intuitive for users I believe - having it above the incoming list makes a little confusing:

![image](https://github.com/user-attachments/assets/ddcfc46a-0d9f-43d2-bec6-f2f2dff3a230)

Reply:
in fact.. it might be nice to wrap this into a ParameterGroup to clarify they go together
![image](https://github.com/user-attachments/assets/7ed79a36-29b4-44df-ac7b-4d10d220d9f8)